### ISSUE TITLE
Medusa verify fails with Blob different error when SSTable file changes in size

### ISSUE THREAD

**Issue by venkata-naladala** [thelastpickle/cassandra-medusa#576]
[Project board link](https://github.com/orgs/k8ssandra/projects/8/views/1?pane=issue&itemId=26732880)

Medusa version: 0.13.4
Apache Cassandra: 4.0.6

Medusa verify fails with the below error. I have renamed the keyspace, table, and IP of the node. I have only included details for one of the hash miss matches.

`medusa verify --backup-name 2023040604`

```
[2023-04-10 14:06:03,531] INFO: Found credentials in shared credentials file: /etc/medusa/credentials
Validating 2023040604 ...
- Completion: OK!
[2023-04-10 14:06:19,719] ERROR: Expected {'path': 'IP/data/keyspace/table/nb-154-big-Statistics.db', 'MD5': 'fa461212848e162fa84adf24aa4745dd', 'size': 5795} but got <Object: name=IP/data/keyspace/table/nb-154-big-Statistics.db, size=5779, hash=2b86d0cae7f359e77396941f126890cb, provider=Driver for all S3 compatible Medusa storages ...>
- Manifest validation: Failed!
 - [IP/data/keyspace/table/nb-154-big-Statistics.db] Blob different
Traceback (most recent call last):
 File "/usr/local/bin/medusa", line 8, in <module>
   sys.exit(cli())
 File "/usr/local/lib/python3.6/site-packages/click/[core.py](https://core.py/)", line 1128, in __call__
   return self.main(*args, **kwargs)
 File "/usr/local/lib/python3.6/site-packages/click/[core.py](https://core.py/)", line 1053, in main
   rv = self.invoke(ctx)
 File "/usr/local/lib/python3.6/site-packages/click/[core.py](https://core.py/)", line 1659, in invoke
   return _process_result(sub_ctx.command.invoke(sub_ctx))
 File "/usr/local/lib/python3.6/site-packages/click/[core.py](https://core.py/)", line 1395, in invoke
   return ctx.invoke(self.callback, **ctx.params)
 File "/usr/local/lib/python3.6/site-packages/click/[core.py](https://core.py/)", line 754, in invoke
   return __callback(*args, **kwargs)
 File "/usr/local/lib/python3.6/site-packages/click/[decorators.py](https://decorators.py/)", line 84, in new_func
   return ctx.invoke(f, obj, *args, **kwargs)
 File "/usr/local/lib/python3.6/site-packages/click/[core.py](https://core.py/)", line 754, in invoke
   return __callback(*args, **kwargs)
 File "/usr/local/lib/python3.6/site-packages/medusa/[medusacli.py](https://medusacli.py/)", line 294, in verify
   medusa.verify.verify(medusaconfig, backup_name, enable_md5_checks)
 File "/usr/local/lib/python3.6/site-packages/medusa/[verify.py](https://verify.py/)", line 55, in verify
   raise RuntimeError("Manifest validation failed")
RuntimeError: Manifest validation failed
```

From the medusa logs, the Statistics.db file size was changed and it was re-uploaded on 2023040704 and this caused the `medusa verify --backup-name 2023040604` to fail.

```
[2023-04-06 04:31:39,066] INFO: Uploading /disk/keyspace/table/snapshots/medusa-2023040604/nb-154-big-Statistics.db (5.659KiB)
[2023-04-07 04:31:44,441] INFO: Uploading /disk/keyspace/table/snapshots/medusa-2023040704/nb-154-big-Statistics.db (5.644KiB)
```

**Comment by adejanovski** [comment]
Actually it's expected that these files can be updated even though sstables are immutable.
Cassandra can redistribute index summaries depending on the number of partitions there is in a table. That can rewrite the `-Statistics` file, which will get a different size and be reuploaded despite having the same name.
This invalidates the previous backups.

The fix here is to ignore the `-Statistics` files from backup verifications, to avoid such breakage.
Restoring an older backup with a different `-Statistics` file than the original one will not break the SSTable nor Cassandra, it will only have a potential impact on performance for as long as the index is recomputed (and that even may not be the case at all).

**Comment by adejanovski** [comment]
This was fixed in 0.15.0 through #587
